migration/sql-files/034-wit-to-use-uuid.sql

```diff
@@ -0,0 +1,107 @@
+CREATE EXTENSION IF NOT EXISTS "ltree";
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+------------------------------------------------------------------------------
+-- Update the work item type table itself:
+--
+-- 0. In parallel to the current primary key ("name" column), we'll add a column
+-- "id" that will become the new primary key later down the road.
+--
+-- 1. For all system-defined WITs, do not use the UUID as it was generated
+-- during the migration instead use the one that is defined in the code.
+--
+-- 2. Add new "description" column and fill with the default value of 'This is
+-- the description for the work item type "X".'.
+--
+-- 3. Update the "path" column of the WIT table to use the new UUID (with "-"
+-- replaced by "_") instead of the "name" column.
+--
+-- 4. Drop the constraint that limits the "name" column to be contain only
+-- C-LOCALE characters. This is a human readable free form field now.
+--
+-- 5. Finally, switch to "id" column to be our new primary key.
+-------------------------------------------------------------------------------
+
+ALTER TABLE work_item_types ADD COLUMN id uuid DEFAULT uuid_generate_v4() NOT NULL;
+
+-- Use WIT IDs define in the code
+UPDATE work_item_types SET id = '{{index . 0}}' WHERE name = 'planneritem';
+UPDATE work_item_types SET id = '{{index . 1}}' WHERE name = 'userstory';
+UPDATE work_item_types SET id = '{{index . 2}}' WHERE name = 'valueproposition';
+UPDATE work_item_types SET id = '{{index . 3}}' WHERE name = 'fundamental';
+UPDATE work_item_types SET id = '{{index . 4}}' WHERE name = 'experience';
+UPDATE work_item_types SET id = '{{index . 5}}' WHERE name = 'feature';
+UPDATE work_item_types SET id = '{{index . 6}}' WHERE name = 'scenario';
+UPDATE work_item_types SET id = '{{index . 7}}' WHERE name = 'bug';
+
+ALTER TABLE work_item_types ADD COLUMN description text;
+UPDATE work_item_types SET description = concat('This is the description for the work item type "', name, '".');
+
+CREATE OR REPLACE FUNCTION UUIDToLtreeNode(u uuid, OUT node ltree) AS $$ BEGIN
```

Comment:
It is not used after that call and I've intentionally not deleted it because I thought we might need this function in the future when doing conversions between Ltree and UUIDs on the DB level.